data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/AbstractLogstashPluginAttributesMapper.java

```diff
@@ -36,15 +36,24 @@
                 .stream()
                 .filter(logstashAttribute -> !customMappedAttributeNames.contains(logstashAttribute.getAttributeName()))
                 .forEach(logstashAttribute -> {
-            if (mappedAttributeNames.containsKey(logstashAttribute.getAttributeName())) {
-                pluginSettings.put(
-                        mappedAttributeNames.get(logstashAttribute.getAttributeName()),
-                        logstashAttribute.getAttributeValue().getValue()
-                );
-            }
-            else {
-                LOG.warn("Attribute name {} is not found in mapping file.", logstashAttribute.getAttributeName());
-            }
+                    final String logstashAttributeName = logstashAttribute.getAttributeName();
+
+                    if (mappedAttributeNames.containsKey(logstashAttributeName)) {
+                        if (mappedAttributeNames.get(logstashAttributeName).startsWith("!")) {
+                            pluginSettings.put(
+                                    mappedAttributeNames.get(logstashAttributeName).substring(1),
+                                    !(Boolean) logstashAttribute.getAttributeValue().getValue()
```

Comment:
It might be a bit cleaner to verify that the attribute is actually a boolean before casting. So something like 

```
if (mappedAttributeNames.get(logstashAttributeName).startsWith("!") && logstashAttribute.getAttributeValue().getValue() instanceOf Boolean)
```

Reply:
Good call. I also think that a new unit test case would be useful here.